enhancements/cguConditionEnhancement.md

```diff
@@ -0,0 +1,527 @@
+[
+  {
+    "lastTransitionTime": "2022-03-21T14:38:09Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "False",
+    "type": "Degraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:39:58Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:43:16Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "True",
+    "type": "Available"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:38:08Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "True",
+    "type": "Upgradeable"
+  }
+]
+```
+
+- **Network operator** has below condition:
+
+```yaml
+[root@hv2 ~]# oc get co/network -ojson | jq '.status.conditions'
+[
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "Degraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "ManagementStateDegraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "True",
+    "type": "Upgradeable"
+  },
+  {
+    "lastTransitionTime": "2022-06-07T15:35:25Z",
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:37:57Z",
+    "status": "True",
+    "type": "Available"
+  }
+]
+```
+
+## Proposal
+
+This section proposes minimalistic changes without breaking the current design pattern. The below proposal/recommendation consists of more granular `status.conditions`  based on all the verification we already have in TALM.
+
+**Considering**: backup,precaching is true and enable for upgrade is true as well. 
+
+### Workflow Description
+
+1. Condition type related to cluster existing in the managedcluster. The type could reflect on the existence of the targeted cluster to be present in the managedcluster. 
+
+  ```yaml
+  Type:    "ClustersSelected",
+  Status:  True/False,
+  Reason:  "",
+  Message: “”
+  ```
+
+2. a) Spec validation for the `managed policies`. Currently it gets checked both during pre-caching and for upgrade. This check can be done before starting an upgrade, irrespective of pre-caching being true or false. b) Platform related verification. 
+Point a) and b) can be merged into **Validated** (previously proposed as PreUpgradeValidation/Upgradeable) where we check managed policies existence, conformity of platform image etc. to understand if upgrade can proceed.
+
+ ```yaml
+  Type:    "Validated",
+  Status:  True/False,
+  Reason:  "MissingManagedPolicy/InvalidPlatformImage/ValidationComplete",
+  Message: “”
+  ```
+
+3. Condition type: **PrecacheSpecValid** and **PrecachingDone**(it will become **PrecachingSucceeded**) which is already in the current design. For **PrecacheSpecValid**, we may omit the highlighted Reason from the [current spec](#precachingSpec)
+
+4. Condition type: **BackupDone** becomes **BackupSucceeded**
+
```

Comment:
Sabbir and I are debating whether backup should be done right after the enable change is made on all chained CGUs or after the blocking CGUs are completed for each CGU. @browsell @imiller0 @donpenney please chime in if you have a preference.

Reply:
After the discussion at the [platform lifecycle working session](https://docs.google.com/document/d/1S4RVy6dRmOTuVtPngB8YM2bUp4CUED_370XZJKPgoPo/edit#heading=h.mlotw7x2vxvw) meeting we agreed to start the backup after blocking CRs are completed in chained CGU case.